cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandlerInternal.java

```diff
@@ -200,4 +215,77 @@ public void getAppDetailForVersion(HttpRequest request, HttpResponder responder,
         : applicationLifecycleService.getAppDetail(appId);
     responder.sendJson(HttpResponseStatus.OK, GSON.toJson(appDetail));
   }
+
+  /**
+   * Deploy an application. This is similar to the public API to deploy an application.
+   * This differs from the public API by supporting the skipMakingLatest parameter.
+   * This behaviour (skipMakingLatest) should not be exposed in the public API and should
+   * only be used internally when needed.
+   *
+   * @param request {@link HttpRequest}
+   * @param responder {@link HttpResponder}
+   * @param namespaceId of the namespace where the app is to be deployed
+   * @param appId of the app
+   * @param skipMarkingLatest if true, the app will be deployed but not marked latest
```

Comment:
nit: Please mention what marking latest means. i.e the latest marked version is the one that will be run

Reply:
done.